neuralprophet/forecaster.py

```diff
@@ -340,6 +340,10 @@ def __init__(
         accelerator: Optional[str] = None,
         trainer_config: dict = {},
     ):
+        kwargs = locals()
+        self.config = kwargs
+        self.config.pop("self")
```

Comment:
What exactly does this line do?

Reply:
it removes the `self` reference from the locals() call, so we do only keep the provided arguments

Reply:
`locals()` returns all locally defined variables (in this case all function parameters)